[export] Implement common_getitem_elimination pass.

zhxchen17 opened the pull request:
Summary:
In export, we will generate many redundant getitem nodes branching from the same source, inserted by runtime assertions or any passes. This is causing issues with any downstream system relying on any value being uniquely defined by a single node.

I don't think it hurt to remove a bunch of getitem nodes only, so I just added to the ctor.

Test Plan:
rebase on D61256937
```
buck2 run scripts/bearzx:pt2_export_playground
```

Differential Revision: D61351578

tugsbayasgalan commented on torch/export/exported_program.py:
```diff
@@ -630,6 +631,28 @@ def update_arg(old_arg, new_ph):
     return gm, new_graph_signature
 
 
+def _common_getitem_elimination_pass(gm: torch.fx.GraphModule, graph_signature):
```
Could you add a test case?

zhxchen17 replied:
added

zhxchen17 committed:
[export] Implement common_getitem_elimination pass. (#133618)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/133618

In export, we will generate many redundant getitem nodes branching from the same source, inserted by runtime assertions or any passes. This is causing issues with any downstream system relying on any value being uniquely defined by a single node.

I don't think it hurt to remove a bunch of getitem nodes only, so I just added to the ctor.

Test Plan:
rebase on D61256937
```
buck2 run scripts/bearzx:pt2_export_playground
```

Differential Revision: D61351578
